cmd/medius/images/verify.go

```diff
@@ -0,0 +1,252 @@
+package images
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	urand "k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/pointer"
+	v1 "kubevirt.io/api/core/v1"
+	kvirtcli "kubevirt.io/client-go/kubecli"
+	kvirtlog "kubevirt.io/client-go/log"
+	"kubevirt.io/containerdisks/cmd/medius/common"
+	"kubevirt.io/containerdisks/pkg/api"
+	"kubevirt.io/containerdisks/pkg/build"
+	"kubevirt.io/containerdisks/pkg/docs"
+	"kubevirt.io/containerdisks/pkg/repository"
+)
+
+const (
+	VerifyUsername = "verify"
+)
+
+func NewVerifyImagesCommand(options *common.Options) *cobra.Command {
+	options.VerifyImagesOptions = common.VerifyImageOptions{
+		Workers:   1,
+		Namespace: "kubevirt",
+		Timeout:   600,
+	}
+
+	verifyCmd := &cobra.Command{
+		Use:   "verify",
+		Short: "Verify that containerdisks are bootable and guests are working",
+		Run: func(cmd *cobra.Command, args []string) {
+			if options.VerifyImagesOptions.ClusterRegistry == "" {
+				options.VerifyImagesOptions.ClusterRegistry = options.Registry
+			}
+
+			// Silence the kubevirt client log
+			kvirtlog.Log = kvirtlog.MakeLogger(kvirtlog.NullLogger{})
+			client, err := kvirtcli.GetKubevirtClient()
+			if err != nil {
+				logrus.Fatal(err)
+			}
+
+			wg, errChan := spawnWorkers(options.VerifyImagesOptions.Workers, options.Focus, func(a api.Artifact) error {
+				return verifyArtifact(a, options, client)
+			})
+			wg.Wait()
+
+			select {
+			case <-errChan:
+				os.Exit(1)
+			default:
+				os.Exit(0)
+			}
+		},
+	}
+	verifyCmd.Flags().IntVar(&options.VerifyImagesOptions.Workers, "workers", options.VerifyImagesOptions.Workers, "Number of parallel workers")
+	verifyCmd.Flags().StringVar(&options.VerifyImagesOptions.ClusterRegistry, "cluster-registry", "", "Registry to use inside the cluster")
+	verifyCmd.Flags().StringVar(&options.VerifyImagesOptions.Namespace, "namespace", options.VerifyImagesOptions.Namespace, "Namespace to run verify in")
+	verifyCmd.Flags().IntVar(&options.VerifyImagesOptions.Timeout, "timeout", options.VerifyImagesOptions.Timeout, "Maximum seconds to wait for VM to be running")
+	verifyCmd.Flags().AddGoFlagSet(kvirtcli.FlagSet())
+
+	return verifyCmd
+}
+
+func verifyArtifact(artifact api.Artifact, options *common.Options, client kvirtcli.KubevirtClient) (err error) {
+	log := common.Logger(artifact)
+
+	imgRefs, err := findImgRefs(artifact, options, log)
+	if err != nil {
+		log.WithError(err).Error("Failed to find containerdisks")
+		return err
+	}
+
+	if len(imgRefs) == 0 {
```

Comment:
Just to double-check, this ensures that we don't fail if some images did not need to be updated, right?

Reply:
Right, this ensures there is at least one image of the `api.Artifact` that needs to be verified.